fetch.ts

```diff
@@ -121,39 +123,52 @@ function firstVal (dataset: DatasetExt) {
   }
 }
 
-function getTitle (dataset: DatasetExt) {
+function getTitle (dataset: DatasetExt): string {
   const potentialValues = [
     firstVal(dataset.match(null, rdf.namedNode(expand('dc11:title')))),
     firstVal(dataset.match(null, rdf.namedNode(expand('dcterms:title')))),
     firstVal(dataset.match(null, rdf.namedNode(expand('rdfs:label'))))
   ].filter(Boolean)
-  return potentialValues.length ? potentialValues[0] : ''
+
+  if (potentialValues.length && potentialValues[0]) {
+    return potentialValues[0]
+  }
+  return ''
 }
 
-function getDescription (dataset: DatasetExt) {
+function getDescription (dataset: DatasetExt): string {
   const potentialValues = [
     firstVal(dataset.match(null, rdf.namedNode(expand('dc11:description')))),
     firstVal(dataset.match(null, rdf.namedNode(expand('dcterms:description')))),
     firstVal(dataset.match(null, rdf.namedNode(expand('rdfs:comment'))))
   ].filter(Boolean)
-  return potentialValues.length ? potentialValues[0] : ''
+
+  if (potentialValues.length && potentialValues[0]) {
+    return potentialValues[0]
+  }
+  return ''
 }
 
 function generateIndex (subject: NamedNode, mappings: any, dataset: DatasetExt) {
   const vocabUri = rdf.namedNode(mappings.uri)
-  const filteredDataset = dataset.match(vocabUri)
+  let filteredDataset = dataset.match(vocabUri)
+  if (vocabUri.value.endsWith('/') || vocabUri.value.endsWith('#')) {
```

Comment:
it did the trick and incidentally also generated many more missing titles and descriptions for other ontologies :)